runtime/near-vm-runner/src/preload.rs

```diff
@@ -41,27 +54,76 @@ pub struct ContractCallPrepareResult {
 
 pub struct ContractCaller {
     pool: ThreadPool,
-    prepared: Vec<CallInner>,
+    vm_kind: VMKind,
+    vm_config: VMConfig,
+    vm_data_private: VMDataPrivate,
+    vm_data_shared: VMDataShared,
+    preloaded: Vec<CallInner>,
 }
 
 impl ContractCaller {
-    pub fn new(num_threads: usize) -> ContractCaller {
-        ContractCaller { pool: ThreadPool::new(num_threads), prepared: Vec::new() }
+    pub fn new(num_threads: usize, vm_kind: VMKind, vm_config: VMConfig) -> ContractCaller {
+        let (shared, private) = match vm_kind {
+            VMKind::Wasmer0 => (
+                VMDataShared::Wasmer0,
+                VMDataPrivate::Wasmer0(if SHARE_MEMORY_INSTANCE {
+                    Some(
+                        WasmerMemory::new(
```

Comment:
This looks confusing, `private = Some(WasmerMemory::new(...))` when it's `SHARE_MEMORY_INSTANCE`, and when not `private = None` 🤔

Reply:
Here `private` means `not-shared-between-threads`.